### DISCUSSION TITLE
Error: Cannot refetch a query that has not been started yet.

### DISCUSSION THREAD

**Discussion by compojoom** [reduxjs/redux-toolkit#4550] [Q&A]
Hey guys,

I have this piece of code
```
  let {
    data: txDetails,
    error: txDetailsError,
    refetch,
  } = useGetTransactionDetailsQuery(
    transactionId && safe.chainId
      ? {
          chainId: safe.chainId,
          txId: transactionId,
        }
      : skipToken,
  )

  useEffect(() => {
    refetch()
  }, [safe.txHistoryTag, safe.txQueuedTag, safeAddress, refetch])
```
and I noticed that it causes an error: `Error: Cannot refetch a query that has not been started yet.`

I understand that it seems that I have no transactionId and no safe.chainId and because of this my query ends up being uninitialized. So I modified it to this:
```
    let {
    data: txDetails,
    error: txDetailsError,
    refetch,
    isUninitialized
  } = useGetTransactionDetailsQuery(
    transactionId && safe.chainId
      ? {
          chainId: safe.chainId,
          txId: transactionId,
        }
      : skipToken,
  )

  useEffect(() => {
    !isUninitialized && refetch()
  }, [safe.txHistoryTag, safe.txQueuedTag, safeAddress, refetch, isUninitialized])

```

and it works fine. However that feels a bit verbose & unexpected? I honestly would have expected that this would just ignore the fetch? I mean if the reason for failure is because I've told rtkquery to skip the query unless there is transctionId && chainId, why wouldn't it do the same when I have a refetch call?

**Comment by EskiMojo14** [comment]
That change was made here: https://github.com/reduxjs/redux-toolkit/pull/2212

Without the check that throws an error, anyone using the return value from `refetch` would have to check whether it was undefined first.

Out of curiosity, why are you calling refetch in useEffect at all? The hook should automatically fetch your data when its argument changes, as long as it's not skipToken.

**Comment by compojoom** [comment]
aha, I see. I call it in useEffect because of the dependencies. `safe.txHistoryTag, safe.txQueuedTag, safeAddress` - the query doesn't depend on those, but I want it to update if any of those changes.
